projects/vdk-core/tests/vdk/internal/builtin_plugins/connection/test_connection_router.py

```diff
@@ -45,7 +53,7 @@ def test_router_raw_connection():
     router.add_open_connection_factory_method("RAW_DB", lambda: mock_conn)
 
     conn = router.open_connection("RAW_DB")
-    assert mock_conn == conn.connect()
+    assert mock_conn == conn._connect()
```

Comment:
why going for non-PEP249 interface here?

Reply:
The test was wrong - the public .connect() method returns Managed Cursor 

The internal _connect method return a raw cursor, and still it's good idea to test it (since it is used by other internal vdk modules).

Reply:
The protected method `_connect` returns native (or raw) cursor.
The public method `connect` returns managed cursor.

Reply:
So what's the reason for that change from a public interface, to invoking a protected native (raw) cursor?

Reply:
There's no change in interfaces. It's a bug fix in the test. But I will remove it from this PR. I should have opened a separate PR (as it is per [our recommendation](https://github.com/vmware/versatile-data-kit/wiki/How-to-prepare-a-new-PR#break-your-code-commits-into--small-self-contained-units) ) . 

See PR https://github.com/vmware/versatile-data-kit/pull/841 for the change and more details.